File: javascript/queue.js
let array = [];
let start = 0;
let end = 0;

const enqueue = (value) => {
  if (end == array.length() - 1) {
    console.log("Queue overflow");
    return;
  }
  array[end] = value;
  end++;
};

const dequeue = () => {
  if (start == end) {
    console.log("Queue underflow");
    return;
  }
  let dequeueValue = array[start];
  start++;
  return dequeueValue;
};

const printArray = () => {
  console.log("printing arrays");
  for (let i = start; i <= end; i++) {
    console.log(array[i]);
  }
};

File: python/queue.py
array = []
start = 0
end = 0


def inputValue():
    value = int(input("Enter value: "))
    return value


def enqueue():
    global end
    if (end == len(array) - 1):
        print("queue overflow")
    array.append(inputValue())
    end += 1


def dequeue():
    global start
    global end
    if (start == end):
        print("queue underflow")
    dequeuedValue = array[start]
    start += 1
    return dequeuedValue


def printQueue():
    global start
    global end
    if (start == end):
        print("queue underflow")
    for i in range(start, end):
        print(array[i])
